src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs

```diff
@@ -524,5 +557,53 @@ private EditingOrderViewModel GetEditingOrderViewModel()
                 })
             };
         }
+
+        public async Task<OrderFulfillmentState> CancelOrder(CancelOrderParam param)
+        {
+            if(param == null) throw new ArgumentNullException(nameof(param));
+            if (param.CultureInfo == null) throw new ArgumentException(GetMessageOfNull(nameof(param.CultureInfo)));
+            if (string.IsNullOrWhiteSpace(param.OrderNumber)) throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.OrderNumber)));
+            if (string.IsNullOrWhiteSpace(param.Scope)) throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.Scope)));
+            if (param.CustomerId == Guid.Empty) throw new ArgumentException(GetMessageOfEmpty(nameof(param.CustomerId)));
+
+            var order = await OrderRepository.GetOrderAsync(new GetCustomerOrderParam
+            {
+                Scope = param.Scope,
+                OrderNumber = param.OrderNumber,
+                CustomerId = param.CustomerId
+            }).ConfigureAwait(false);
+            var orderId = Guid.Parse(order.Id);
+            if (order == null) throw new InvalidOperationException($"Order {param.OrderNumber} cannot be received.");
+
+            var isOrderCancelable = await EditingOrderProvider.IsOrderCancelable(order).ConfigureAwait(false);
+            if(!isOrderCancelable) throw new InvalidOperationException($"For canceling order {param.OrderNumber} should be cancelable.");
+            
+            if (string.IsNullOrWhiteSpace(order.OrderStatus)) throw new InvalidOperationException($"Order {param.OrderNumber} has no status.");
+            
+            var shipment = order.Cart?.Shipments?.FirstOrDefault();
+            if (shipment?.Id == null) throw new InvalidOperationException($"Shipment for the order {param.OrderNumber} is null.");
+
+            await OrderRepository.ChangeShipmentStatusAsync(new ChangeShipmentStatusParam
```

Comment:
In EditingOrderProvider add method CancelOrder(Overture.ServiceModel.Orders.Order order) 
inside the method do validation if Order and FulfilmentState can be canceled then execute change status

call this method from this service.